SU2_CFD/include/variables/CMeshElement.hpp

```diff
@@ -0,0 +1,108 @@
+/*!
+ * \file CMeshElement.hpp
+ * \brief Declaration and inlines of the class
+ *        to define the variables of the mesh movement.
+ * \author Ruben Sanchez
+ * \version 6.2.0 "Falcon"
+ *
+ * The current SU2 release has been coordinated by the
+ * SU2 International Developers Society <www.su2devsociety.org>
+ * with selected contributions from the open-source community.
+ *
+ * The main research teams contributing to the current release are:
+ *  - Prof. Juan J. Alonso's group at Stanford University.
+ *  - Prof. Piero Colonna's group at Delft University of Technology.
+ *  - Prof. Nicolas R. Gauger's group at Kaiserslautern University of Technology.
+ *  - Prof. Alberto Guardone's group at Polytechnic University of Milan.
+ *  - Prof. Rafael Palacios' group at Imperial College London.
+ *  - Prof. Vincent Terrapon's group at the University of Liege.
+ *  - Prof. Edwin van der Weide's group at the University of Twente.
+ *  - Lab. of New Concepts in Aeronautics at Tech. Institute of Aeronautics.
+ *
+ * Copyright 2012-2019, Francisco D. Palacios, Thomas D. Economon,
+ *                      Tim Albring, and the SU2 contributors.
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "../../../Common/include/mpi_structure.hpp"
+
+#include <cmath>
+#include <iostream>
+#include <cstdlib>
+
+#include "../../../Common/include/config_structure.hpp"
+
+class CMeshElement {
```

Comment:
If one day this class needs to start keeping dynamic data I would say now is the time to implement a contiguous storage approach.

Reply:
Yep I think in the long run #753 is the way to go. In principle this class is not intended to store dynamic data, but you never know. I still need to go over #753 in more detail to understand completely the logic of the implementation.